### PR TITLE
Fix PAPI wrapper

### PR DESCRIPTION
**Pull request by vchuravy**

### CODE REVIEW

File: P/PAPI/build_tarballs.jl [hunk 1]
```diff
@@ -109,14 +109,13 @@ for cuda_version in cuda_versions_to_build, platform in platforms
...
+    ]
     if cuda_version != "none"
         if arch(platform) in cuda_platforms
-            dependencies = [
-                BuildDependency(PackageSpec(name="CUDA_full_jll",
-                                            version=cuda_full_versions[cuda_version])),
-                RuntimeDependency(PackageSpec(name="CUDA_Runtime_jll")),
-            ]
+		push!(dependencies, BuildDependency(PackageSpec(name="CUDA_full_jll",
+								version=cuda_full_versions[cuda_version])))
```
Comment (vchuravy) [inline P/PAPI/build_tarballs.jl:118]:
```suggestion
        push!(dependencies, BuildDependency(PackageSpec(name="CUDA_full_jll",
                                                                                               version=cuda_full_versions[cuda_version])))
```

File: P/PAPI/build_tarballs.jl [hunk 2]
```diff
@@ -109,14 +109,13 @@ for cuda_version in cuda_versions_to_build, platform in platforms
                                   cuda=tag)
     should_build_platform(triplet(augmented_platform)) || continue
 
-    dependencies = []
+    dependencies = Dependency[
```
Comment (giordano) [inline P/PAPI/build_tarballs.jl:112]: Did you mean AbstractDependency? Otherwise I think you're converting the runtime dep to normal dep

Comment (giordano) [reply P/PAPI/build_tarballs.jl:112]: And in fact CUDA_Runtime_jll is installed in the build environment: https://buildkite.com/julialang/yggdrasil/builds/1024#018665b2-a065-493b-9acd-c844ee393ebc/637-651 and below there's a warning about no artifact being available.

Comment (vchuravy) [inline P/PAPI/build_tarballs.jl:112]:
```suggestion
    dependencies = AbstractDependency[
```

File: platforms/cuda.jl [hunk 3]
```diff
@@ -16,7 +16,7 @@ const augment = """
 
     function cuda_comparison_strategy(a::String, b::String, a_requested::Bool, b_requested::Bool)
         if a == "none" || b == "none"
-            return false
+            return a == b
         end
```
Comment (vchuravy) [inline platforms/cuda.jl:20]: cc: @maleadt any opposition to this change?

Comment (maleadt) [reply platforms/cuda.jl:20]: Took me a while, but I think this is fine.

File: P/PAPI/build_tarballs.jl [hunk 4]
```diff
@@ -109,14 +109,13 @@ for cuda_version in cuda_versions_to_build, platform in platforms
...
+        RuntimeDependency(PackageSpec(name="CUDA_Runtime_jll")),
+    ]
     if cuda_version != "none"
         if arch(platform) in cuda_platforms
-            dependencies = [
-                BuildDependency(PackageSpec(name="CUDA_full_jll",
-                                            version=cuda_full_versions[cuda_version])),
-                RuntimeDependency(PackageSpec(name="CUDA_Runtime_jll")),
-            ]
+        push!(dependencies, BuildDependency(PackageSpec(name="CUDA_full_jll",
```
Comment (maleadt) [inline P/PAPI/build_tarballs.jl:117]: Wrong indentation

**Review by maleadt — APPROVED** [review]

**Pull request merged**